src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs

```diff
@@ -0,0 +1,137 @@
+            var testOptions = TestFrameworkOptions.ForExecution(configuration);
+            var testSink = new TestMessageSink();
+            var controller = new Xunit2(AppDomainSupport.Denied, new NullSourceInformationProvider(), assemblyFileName, configFileName: null, shadowCopy: false, shadowCopyFolder: null, diagnosticMessageSink: diagnosticSink, verifyTestAssemblyExists: false);
+
+            discoveryOptions.SetSynchronousMessageReporting(true);
+            testOptions.SetSynchronousMessageReporting(true);
+
+            Console.WriteLine($"Discovering tests for {assemblyFileName}");
+            var assembly = Assembly.LoadFrom(assemblyFileName);
+            var assemblyInfo = new global::Xunit.Sdk.ReflectionAssemblyInfo(assembly);
+            var discoverer = new ThreadlessXunitDiscoverer(assemblyInfo, new NullSourceInformationProvider(), discoverySink);
+
+            discoverer.FindWithoutThreads(includeSourceInformation: false, discoverySink, discoveryOptions);
+            discoverySink.Finished.WaitOne();
+            var testCasesToRun = discoverySink.TestCases.Where(filters.Filter).ToList();
+            Console.WriteLine($"Discovery finished.");
+
+            var summarySink = new DelegatingExecutionSummarySink(testSink, () => false, (completed, summary) => { Console.WriteLine($"Tests run: {summary.Total}, Errors: 0, Failures: {summary.Failed}, Skipped: {summary.Skipped}{Environment.NewLine}Time: {TimeSpan.FromSeconds((double)summary.Time).TotalSeconds}s"); });
+            var resultsXmlAssembly = new XElement("assembly");
+            var resultsSink = new DelegatingXmlCreationSink(summarySink, resultsXmlAssembly);
+
+            testSink.Execution.TestPassedEvent += args => { Console.WriteLine($"[PASS] {args.Message.Test.DisplayName}"); };
+            testSink.Execution.TestSkippedEvent += args => { Console.WriteLine($"[SKIP] {args.Message.Test.DisplayName}"); };
+            testSink.Execution.TestFailedEvent += args => { Console.WriteLine($"[FAIL] {args.Message.Test.DisplayName}{Environment.NewLine}{ExceptionUtility.CombineMessages(args.Message)}{Environment.NewLine}{ExceptionUtility.CombineStackTraces(args.Message)}"); };
+
+            testSink.Execution.TestAssemblyStartingEvent += args => { Console.WriteLine($"Running tests for {args.Message.TestAssembly.Assembly}"); };
+            testSink.Execution.TestAssemblyFinishedEvent += args => { Console.WriteLine($"Finished {args.Message.TestAssembly.Assembly}{Environment.NewLine}"); };
+
+            controller.RunTests(testCasesToRun, resultsSink, testOptions);
+            resultsSink.Finished.WaitOne();
+
+            if (printXml)
+            {
+                Console.WriteLine($"STARTRESULTXML");
+                var resultsXml = new XElement("assemblies");
+                resultsXml.Add(resultsXmlAssembly);
+                Console.WriteLine(resultsXml.ToString());
+                Console.WriteLine();
+                Console.WriteLine($"ENDRESULTXML");
```

Comment:
In the other runners, we write the xml to a file if we want and to the logger, same as the previous reason, lets try to have all of the them write into a logger that can later abstract the tranport.